server/routes/api/team.ts

```diff
@@ -60,4 +62,54 @@ router.post("team.update", auth(), async (ctx) => {
   };
 });
 
+router.post("team.create", auth(), async (ctx) => {
+  const { user } = ctx.state;
+  const { name } = ctx.body;
+  assertLength(name, 2, "Name must be 2 or more characters");
+
+  const existingTeam = await Team.scope("withAuthenticationProviders").findByPk(
+    user.teamId
+  );
+
+  const authenticationProviders = existingTeam?.authenticationProviders.map(
+    (provider) => {
+      return {
+        name: provider.name,
+        providerId: provider.providerId,
+      };
+    }
+  );
+
+  invariant(
+    authenticationProviders?.length,
+    "authentication provideers must exist"
+  );
+
+  const team = await teamCreator({
+    name,
+    subdomain: name,
+    authenticationProviders,
+    ip: ctx.ip,
+  });
+
+  await User.create({
+    teamId: team.id,
+    name: user.name,
+    email: user.email,
+    service: null,
+    isAdmin: true,
+    avatarUrl: user.avatarUrl,
+    invitedById: user.id,
+  });
+
+  // make an event here?
```

Comment:
@tommoor should we make an event here? it's not reeeally a user being created, but it might be good for... consistency?